documentation/documentation_publisher.ts

```diff
@@ -0,0 +1,110 @@
+import AWS from 'aws-sdk';
+import S3 from 'aws-sdk/clients/s3';
+import {promisify} from 'util';
+import {print, printAndExit} from '../testing/helpers';
+import {version} from '../package.json';
+import yargs, {Arguments} from 'yargs';
+import {exec as childExec} from 'child_process';
+const exec = promisify(childExec);
+
+const AwsRegion = 'us-west-2';
+const BaseGeneratedDocsPath = 'site';
+const DocumentationBucket = 'developer-documentation';
+const PacksSdkBucketRootPath = 'packs';
+const EnvironmentKeyRoot = `${PacksSdkBucketRootPath}/current`;
+const Environments = ['adhoc', 'head', 'staging', 'prod'];
+
+function handleError(e: Error) {
+  printAndExit(e.message + ` ${e.stack || ''}`);
+}
+
+function getS3Service(env: string): S3 {
+  const config: S3.ClientConfiguration = {
+    signatureVersion: 'v4',
+    region: AwsRegion,
+    credentials: new AWS.SharedIniFileCredentials({profile: env}),
```

Comment:
Have you been able to test this locally? If so, is there a way to print a failure message if the credentials are invalid/missing, that explains what a developer should do to set up their environment to test this?

Reply:
yeah i've been testing this locally and sure thing! I believe `prodaccess-admin` inserts the appropriate credentials into the ~/.aws/credentials file so i can try once my credentials are expired to see what the error is